CONTRIBUTING.md

```diff
@@ -17,15 +17,63 @@ try to keep the codebase consistent and clean.
 
 ## Getting Started
 
-* Make sure you have a GitHub account.
+* Make sure you have a GitHub account and solid ability to use `git`.
 * Fork the repository on GitHub.
-* Create a topic branch from master. Please avoid working directly on the ```master``` branch.
+* Create a topic branch from master. Please avoid working directly on the `master` branch.
 * Make commits of logical units.
-* Check for coding style issues with ```git diff master..mybranch | ./sys/clang-format-diff.py -p1``` and be sure to follow the CODINGSTYLE (more on this in [DEVELOPERS.md](https://github.com/radareorg/radare2/blob/master/DEVELOPERS.md)).
+* Check for coding style issues with:
+
+      git diff master..mybranch | ./sys/clang-format-diff.py -p1
+
+  and be sure to follow the CODINGSTYLE (more on this in [DEVELOPERS.md][]).
 * Submit the Pull Request(PR) on Github.
 * When relevant, write a test in [test/](test).
 
+## Rebasing onto updated master
+
+Every so often, your PR will lag behind `master` and get conflicts.
+
+To "update" your branch `my-awesome-feature`, you *rebase* it onto
+the latest `radareorg/master`, and *force-push* the result into your fork.
+
+Be sure to understand *git remotes* as you need working with 3 git repos
+simultaneously: 1: your local clone, 2: radareorg GH repo, 3: your GH fork.
+
+For example, user Joe rebases his `my-awesome-feature` and using
```

Comment:
why do you need `git remote rename` and `git remote set-url`? I think mentioning them just creates confusion as I've never used them for r2.

Reply:
for me this should be simplified by saying: 

1. Fork the radareorg/radare2 repository
2. Clone the forked repository
3. Create a new branch
4. Do the changes and commit them
5. Push everything on the forked repository
6. Open a PR and wait to be merged
7. Once merged, checkout to master
8. Add remote to radareorg/radare2 repo (step that needs to be done only once)
9. Git fetch, rebase and push

additional steps in case of needed rebase on the PR

1. Add remote to radareorg/radare2 repo (step that needs to be done only once)
2. Git fetch
3. Rebase interactively and fix any conflicts
4. Git push `-f`

Reply:
@ret2libc because people get confused about what `origin` is; when cloning the fork-repo, `origin` won't point to the "logical" origin -- it will point to the fork. As I'm describing specifically the case of conflicts with master: adding the second `radareorg` remote will be unavoidable. For clueless folks, it gets really messy real quick; that's why I'm pointing out the basics of managing remotes, and (indirectly) what they even are.

You didn't hit this stuff — simply because you're collaborator. You can push branches directly to `radareorg`, and need no fancy remote manipulations. Outside folks do need to deal with that; see #16621.

@wargio my list has 3 steps, yours is 9... in what sense is that simplification? :)
Please notice the section header and 1st paragraph. No need to duplicate the already existing "Getting started" thing. This section is *specifically* for rebasing old PRs with conflicts.

Reply:
> You didn't hit this stuff — simply because you're collaborator. You can push branches directly to `radareorg`, and need no fancy remote manipulations. Outside folks do need to deal with that; see #16621.

Even though I'm a collaborator I hardly, if ever, push directly to branches. I do work with multiple remotes: mine, radareorg one, several other contributors. But I never used `git remote rename` or never had to use `git remote set-url` (said in other words, I did use them, but for doing very different stuff. And if a user needs to read this basic git stuff, I think mentioning these things would just confuse them even more, because they are probably not very familiar with git in the first place).

Reply:
@ulidtko i actually always push to my fork, even if i'm a collab. and all these steps allows people who are not used to git to understand each step.

Reply:
My steps use to be like this:

* git checkout -b my-cool-feature
* do changes in code and commit
* git checkout master
* git pull https://github.com/radareorg/radare2
* git checkout my-cool-feature
* git rebase master

Reply:
without remotes or origins, its more implicit and may be less confusing for the reader

Reply:
Ah, I often forget that git accepts full URLs in place of `remote` args.

Also yes, updating local master should be part of the steps, as not doing it (and not saying explicitly `origin/master`) is very often made mistake.